### ISSUE TITLE
[inductor] RuntimeError: CUDA error: an illegal memory access was encountered when using max-autotune

### ISSUE THREAD

**Issue by parthmannan** [pytorch/pytorch#95335]
### 🐛 Describe the bug

Trying to use Triton generated matmuls leads to the error.

### Error logs

```
[2023-02-22 23:05:28,592] torch._inductor.graph: [ERROR] Error from lowering
Traceback (most recent call last):
  File "/opt/pytorch/pytorch/torch/_inductor/graph.py", line 342, in call_function
    out = lowerings[target](*args, **kwargs)
  File "/opt/pytorch/pytorch/torch/_inductor/lowering.py", line 226, in wrapped
    out = decomp_fn(*args, **kwargs)
  File "/opt/pytorch/pytorch/torch/_inductor/kernel/mm.py", line 142, in tuned_addmm
    return autotune_select_algorithm(choices, [inp_expanded, mat1, mat2], layout)
  File "/opt/pytorch/pytorch/torch/_inductor/select_algorithm.py", line 597, in __call__
    timings = self.lookup(
  File "/opt/pytorch/pytorch/torch/_inductor/codecache.py", line 166, in lookup
    benchmark(choice),
  File "/opt/pytorch/pytorch/torch/_inductor/select_algorithm.py", line 573, in autotune
    timing = benchmark_fn(
  File "/opt/pytorch/pytorch/torch/_inductor/select_algorithm.py", line 638, in benchmark
    result = do_bench(lambda: algo(*example_inputs, out=out))
  File "/usr/local/lib/python3.10/site-packages/triton/testing.py", line 144, in do_bench
    torch.cuda.synchronize()
  File "/opt/pytorch/pytorch/torch/cuda/__init__.py", line 702, in synchronize
    return torch._C._cuda_synchronize()
RuntimeError: CUDA error: an illegal memory access was encountered
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.
```

The complete error log is attached.
[debug_ind_mm_error.log](https://github.com/pytorch/pytorch/files/10809024/debug_ind_mm_error.log)

### Minified repro

```
import torch
import torch.nn as nn
import torch.nn.functional as F

class Net(nn.Module):
    def __init__(self, config):
        super(Net, self).__init__()
        self.fc1 = nn.Linear(config['input_units'], config['hidden_units'], bias=True)
        
    def forward(self, x):
        out      = self.fc1(x)
        
        return out

device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")

config = {'input_units': 768, 'hidden_units': 768, 'seq_len': 512, 'batch_size': 64}

net = Net(config)
net.cuda()
net = net.half()

input_shape    = [config['seq_len'], config['batch_size'], config['input_units']]
target_shape   = [config['seq_len'], config['batch_size'], config['hidden_units']]

network_fn = torch.compile(net, mode="max-autotune")
#network_fn = torch.compile(net)

bench_iters = 8
profile_batch = 5

for idx in range(bench_iters):

    inp_tensor = torch.rand(input_shape, dtype=torch.float16, requires_grad=True, device='cuda')
    tgt_tensor = torch.rand(target_shape, dtype=torch.float16, requires_grad=True, device='cuda')
    
    outputs = network_fn(inp_tensor)
    outputs.backward(tgt_tensor)
```

### Versions

```
Collecting environment information...
PyTorch version: 2.0.0a0+git1ab112c
Is debug build: False
CUDA used to build PyTorch: 12.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.5 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.25.2
Libc version: glibc-2.31

Python version: 3.10.9 (main, Feb 21 2023, 00:37:48) [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-4.15.0-144-generic-x86_64-with-glibc2.31
Is CUDA available: True
CUDA runtime version: 12.1.66
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA A100 80GB PCIe
Nvidia driver version: 525.85.12
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.8.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.8.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.8.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.8.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.8.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.8.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.8.0
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   43 bits physical, 48 bits virtual
CPU(s):                          16
On-line CPU(s) list:             0-15
Thread(s) per core:              2
Core(s) per socket:              8
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       AuthenticAMD
CPU family:                      23
Model:                           49
Model name:                      AMD EPYC 7232P 8-Core Processor
Stepping:                        0
Frequency boost:                 enabled
CPU MHz:                         1719.890
CPU max MHz:                     3100.0000
CPU min MHz:                     1500.0000
BogoMIPS:                        6199.80
Virtualization:                  AMD-V
L1d cache:                       256 KiB
L1i cache:                       256 KiB
L2 cache:                        4 MiB
L3 cache:                        32 MiB
NUMA node0 CPU(s):               0-15
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Full AMD retpoline, IBPB conditional, IBRS_FW, STIBP conditional, RSB filling
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid extd_apicid aperfmperf pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate sme ssbd ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif umip rdpid overflow_recov succor smca

Versions of relevant libraries:
[pip3] numpy==1.23.5
[pip3] torch==2.0.0a0+git1ab112c
[pip3] torchvision==0.15.0a0+a192c95
[conda] Could not collect
```

cc @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh

**Comment by parthmannan** [comment]
Looks like similar issue was posted just a few minutes ago as I was typing this issue up - #95312 
Will close this once either gets fixed. Apologies for duplicate.

**Comment by aifartist** [comment]
Yes, it looks like a dup.  @ptrblck asked me to post it.  Looks like someone also posted a very simple repro on my issue.

**Comment by ngimel** [comment]
If you guys can check 1-line fix #95339 and verify that it works for your cases, that would be great

**Comment by parthmannan** [comment]
The fix works for my case. Thanks Natalia for the quick fix.
Closing this one as completed now.
